main.go

```diff
@@ -35,13 +35,16 @@ import (
 
 func main() {
 	var cliConfig string
+	var APIGroup string
 
 	flagSet := flag.NewFlagSet("cluster-machine-approver", flag.ExitOnError)
 
 	klog.InitFlags(flagSet)
 	flagSet.StringVar(&cliConfig, "config", "", "CLI config")
 	flagSet.Parse(os.Args[1:])
 
+	flagSet.StringVar(&APIGroup, "apigroup", "machine.openshift.io", "API group for machines, defaults to machine.openshift.io")
```

Comment:
Do we want to include the kubeconfig versions changes that would be needed for split clusters in this PR or is that a follow up?

Reply:
I'm working on it right now, can be later merged in a follow-up

Reply:
@alexander-demichev can we please validate accepted apigroups and fail otherwise?

Reply:
This comment was resolved without being resolved